source/validate_id.cpp

```diff
@@ -1629,18 +1629,18 @@ bool walkCompositeTypeHierarchy(
     std::vector<uint32_t>::const_iterator word_iter,
     std::vector<uint32_t>::const_iterator word_iter_end,
     const libspirv::Instruction* base,
-    const libspirv::Instruction*& result_type_instr,
-    std::function<std::string(void)> instr_name, std::stringstream& error) {
-  result_type_instr = base;
+    const libspirv::Instruction** result_type_instr,
+    std::function<std::string(void)> instr_name, std::ostringstream* error) {
```

Comment:
std::ostream is the more basic type to use in the callee.  In the callee method your'e only shifting text onto it.